doc/arkode/guide/source/Usage/LSRKStep/User_callable.rst

```diff
@@ -0,0 +1,361 @@
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARKLS_MEM_NULL* if ``arkode_mem`` was ``NULL``.
+      * *ARK_ILL_INPUT* if an argument had an illegal value (e.g. ``nsteps < 0``)
+
+
+.. c:function:: int LSRKStepSetMaxNumStages(void* arkode_mem, int stage_max_limit);
+
+   Specifies the maximum number of stages allowed within each time step.  This bound only applies to
+   RKL and RKC methods.
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *stage_max_limit* -- maximum allowed number of stages :math:`(>1)`.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARKLS_MEM_NULL* if ``arkode_mem`` was ``NULL``.
+      * *ARK_ILL_INPUT* if an argument had an illegal value (e.g. ``stage_max_limit < 2``)
+
+.. note:: If LSRKStepSetMaxNumStages routine is not called, then the default stage_max_limit is
+   set to \sqrt(reltol/(10uround)).
+
+
+.. c:function:: int LSRKStepSetDomEigSafetyFactor(void* arkode_mem, sunrealtype dom_eig_sfty);
+
+   Specifies a safety factor to use for the result of the dominant eigenvalue estimation function.  This value is used to scale the magnitude of the dominant eigenvalue, in the hope of ensuring a sufficient number of stages for the method to be stable.  This input is only used for RKC and RKL methods.
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *dom_eig_sfty* -- safety factor :math:`(\ge 1)`.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARKLS_MEM_NULL* if ``arkode_mem`` was ``NULL``.
+      * *ARK_ILL_INPUT* if an argument had an illegal value (e.g. ``dom_eig_sfty < 1``)
+
+
+.. c:function:: int LSRKStepSetSSPStageNum(void* arkode_mem, int num_of_stages);
+
+   Sets the number of stages, ``s`` in ``SSP(s, p)`` methods. This input is only utilized by SSPRK methods.
+
+      * ``ARKODE_LSRK_SSP_S_2``  -- ``num_of_stages`` must be greater than or equal to 2
+      * ``ARKODE_LSRK_SSP_S_3``  -- ``num_of_stages`` must be a perfect-square greater than or equal to 9
+      * ``ARKODE_LSRK_SSP_10_4`` -- ``num_of_stages`` cannot be modified from 10, so this function should not be called.
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *num_of_stages* -- number of stages :math:`(>1)` for ``SSP(s,2)`` and :math:`(n^2 = s \geq 9)` for ``SSP(s,3)``.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARKLS_MEM_NULL* if ``arkode_mem`` was ``NULL``.
+      * *ARK_ILL_INPUT* if an argument had an illegal value (e.g. ``num_of_stages < 2``)
+
+
+.. _ARKODE.Usage.LSRKStep.OptionalOutputs:
+
+Optional output functions
+------------------------------
+
+
+.. c:function:: int LSRKStepGetNumRhsEvals(void* arkode_mem, int num_rhs_fn, long int* f_evals);
```

Comment:
I think that a number of these comments are irrelevant to this specific PR, and should be discussed elsewhere.

However, in response to this question, the current design [that we agreed upon last Spring] is in place because each stepper supports a potentially custom number of RHS functions (e.g., ERKStep supports 1 while ARKStep supports 2).  Since we cannot overload operators in C to support the same function with differing numbers of outputs, and since only the stepper keeps track of the number of function evals for each of it's own RHS functions, these "GetNumRhsEvals" routines live in the steppers.

Reply:
Since this this function returns an array of RHS evaluations it can cover all the integrators in ARKODE with the steppers providing a function to fill the array (much like how they all provide `printmem`, `setdefaults`, `free`, etc.)

Reply:
I agree, although I am loathe to keep piling duties onto Mustafa that are unrelated to his PR.  Our team has a bad habit of asking PR requestors to make numerous changes to older code, instead of adding those requests to a separate list of group development tasks.

I recommend that some other SUNDIALS developer should add this function to ARKODE, and then add all of the relevant documentation (e.g., for ARKStep `num_rhs_fn=0` corresponds to `fe` while 'num_rhs_fn=1` correspnds to `fi`).  If this gets merged soon, then Mustafa's routine could easily be turned into an internal stepper-provided function for ARKODE to call.

Reply:
I agree with Dan.  Since the other steppers currently have their own `GetNumRhsEvals` functions, this PR is not the right place to add a `ARKodeGetNumRhsEvals`.

Reply:
I'll try to get a PR up for this soon, I'd like to keep churn in the user interface to a minimum.

Reply:
Once PR #587 is merged, we should update this PR to leverage the new approach.